jerry-core/vm/vm.c

```diff
@@ -64,7 +64,7 @@ vm_op_get_value (ecma_value_t object, /**< base object */
 #ifdef JERRY_CPOINTER_32_BIT
       bool limit_check = (int_value >= 0);
 #else /* !JERRY_CPOINTER_32_BIT */
-      bool limit_check = (int_value >= 0 && int_value < (UINT16_MAX + 1));
+      bool limit_check = (int_value >= 0 && (ecma_value_t)int_value < (UINT16_MAX + 1));
```

Comment:
Why would it matter -- the code compares int_value, which is known to fit into 31 bit unsigned integer (since it is a positive signed 32 bit value) to something that already is 17 bit unsigned value, which obviously fits into 32 bit unsigned value.

If this gets in the way of this change being approved I'll get it your way.